automatic_data_generation/data/base_dataset.py

```diff
@@ -271,3 +281,49 @@ def embed_unks(self, init="randn", num_special_toks=2):
             "total number of words are {}"
                 .format(running_norm / num_non_zero, num_non_zero, total_words)
         )
+
+    def save(self, folder):
+        folder = Path(folder)
+        if not folder.exists():
+            folder.mkdir()            
+        vocab_dict = {'i2w':self.i2w, 'i2int':self.i2int}
+        if self.input_type == 'delexicalised':
+            vocab_dict['slotdic'] = self.slotdic
+        torch.save(vocab_dict, folder / "vocab.pth")
+                
+    def update(self, folder):
+
+        folder = Path(folder)
+        loaded_dict    = torch.load(str(folder / "vocab.pth"))
+        loaded_i2w     = loaded_dict['i2w']
+        loaded_i2int   = loaded_dict['i2int']
+        self.update_vocab(self.vocab, loaded_i2w)
+        self.update_vocab(self.intent.vocab, loaded_i2int)
+        self.update_vectors()
+
+        if self.input_type == 'delexicalised':
+            loaded_slotdic = loaded_dict['slotdic']
+            self.update_slotdic(loaded_slotdic)
```

Comment:
it's not ideal because it won't work for any other class than `SnipsDataset`. To remedy this, I would create an abstract method in the Base class that raises a NotImplementedError